plutus-ledger/src/Ledger/Index.hs

```diff
@@ -177,6 +177,7 @@ validateTransaction :: ValidationMonad m
 validateTransaction h t = do
     -- Phase 1 validation
     checkSlotRange h t
+    checkValidInputs (toListOf (inputs . scriptTxIns)) t
```

Comment:
In phase 1 we need to check that the inputs exist, and in phase 2 we run the actual scripts that unlock the inputs. So it's only part of `checkValidInputs` that happens in phase 1 (sorry I was a unclear about this in my earlier comment)